### PR TITLE
Case insensitive command path

### LINKED ISSUE #3325: Command paths should be case insensitive
Example. This works:
  az network dns record-set a show
This fails:
  az network dns record-set A show
That's strange, since DNS record types are normally capitalized!
The record type, in all DNS commands, should be case-insensitive.

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #3325.

1. Command path and command name are no longer case sensitive
2. Long and short options REMAIN case-sensitive

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [N/A] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/application.py [hunk 1]
```diff
@@ -154,13 +154,14 @@ def execute(self, unexpanded_argv):  # pylint: disable=too-many-statements
 
         # Rudimentary parsing to get the command
         nouns = []
-        for noun in argv:
+        for i in range(len(argv)):  # pylint: disable=consider-using-enumerate
             try:
-                if noun[0] == '-':
+                if argv[i][0] == '-':
```
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/application.py]: This checks each char in the noun?
Would this mean I can't have `-` in my command? (e.g. `az vm my-command`)

Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/application.py]: It just checks the first character of the part.  So it has "found" the command once it hits the first things that starts with - (long or short options)

**Comment by tjprescott** [comment]
This also fixes the issue of master branch failing due to merge-corrupted recordings.

File: src/azure-cli-core/azure/cli/core/application.py (hunk 1 above)
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/application.py]: Isn't that what the prev. code did?
Does `noun.lower()` not work?

Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/application.py]: `noun.lower()` creates a lowercase copy of the argv component without actually changing it.  If I try to parse the command line without modifying the argv component, it raises a ParseError.

**Pull request merged**